### PR TITLE
Make platform a proper module

### LINKED ISSUE #16019: tf-nightly and master - cannot import tensorflow
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: 
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16
- **TensorFlow installed from (source or binary)**: Binary
- **TensorFlow version (use command below)**: tf-nightly-gpu-1.6.0.dev20180110
- **Python version**: 2.7
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: 9.0/7
- **GPU model and memory**: V100 16GB
- **Exact command to reproduce**: ```import tensorflow```

### Describe the problem
On the current tf-nightly-gpu I cannot import tensorflow, the following error is produced. I am also seeing the same behavior on tf-nightly and also a build from source of master (SHA: 82b1e8eee8847730026379e3a5762c0e09d6fd36):
``` python
In [1]: import tensorflow
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-64156d691fe5> in <module>()
----> 1 import tensorflow as tf

/home/ubuntu/tensorflow/tensorflow/__init__.py in <module>()
     22
     23 # pylint: disable=wildcard-import
---> 24 from tensorflow.python import *
     25 # pylint: enable=wildcard-import
     26

/home/ubuntu/tensorflow/tensorflow/python/__init__.py in <module>()
     47 import numpy as np
     48
---> 49 from tensorflow.python import pywrap_tensorflow
     50
     51 # Protocol buffers

/home/ubuntu/tensorflow/tensorflow/python/pywrap_tensorflow.py in <module>()
     23 import traceback
     24
---> 25 from tensorflow.python.platform import self_check
     26
     27

ImportError: No module named platform
```

### Source code / logs
N/A

### PR DESCRIPTION
**Pull request by droidicus**
This fixes an issue where the nice error about importing tensorflow from the TF source directory is not displayed in Python 2.7

Fixes #16019

### CODE REVIEW

**Comment by droidicus** [comment]
I signed it!

File: tensorflow/python/eager/test.py [hunk 1]
```diff
@@ -23,6 +23,6 @@
 from tensorflow.python.platform.test import *  # pylint: disable=wildcard-import
 
 
-def main(argv=None):
+def main(argv=None): # pylint: disable=function-redefined
```
Comment (rmlarsen) [inline tensorflow/python/eager/test.py]: Two spaces before '#'

**Review by rmlarsen — APPROVED** [review]

File: tensorflow/python/platform/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (rmlarsen) [inline tensorflow/python/platform/__init__.py]: please revert

**Review by rmlarsen — CHANGES_REQUESTED** [review]

**Comment by rmlarsen** [comment]
@droidicus Is this still needed?

**Comment by droidicus** [comment]
It looks like this has already been fixed on master in another way, closing MR and associated issue.

**Pull request closed without merging**